src/components/LabelButton/LabelButton.jsx

```diff
@@ -32,18 +35,31 @@ function LabelButton({ noteData, setNoteData }) {
             className="text-md"
             onClick={(e) => {
               e.preventDefault();
-              setNoteData({
-                ...noteData,
-                tags:
-                  newLabel !== ""
-                    ? [...noteData.tags, newLabel]
-                    : [...noteData.tags],
-              });
+              modalState
+                ? setModal({
+                    ...modalState,
+                    tags: [...modalState.tags, newLabel],
+                  })
+                : notesDispatch({ type: "UPDATE-TAGS", payload: newLabel });
```

Comment:
label or tags use a single name for better understanding

Reply:
OK